or-monte-carlo/src/com/waterloorocketry/openrocket_monte_carlo/SimulationEngine.java

```diff
@@ -162,44 +159,80 @@ public List<SimulationData> processSimulationData() {
         log.info("Initial stability: {}", initStability);
         log.info("Percentage of initial stability less than 1.5: {}", lowInitStabilityPercentage);
 
-        return data.stream().sorted(Comparator.comparing(SimulationData::getMinStability))
-                .limit(5).toList();
+//        return data.stream().sorted(Comparator.comparing(x -> x.getMinStability().get(0)))
+//                .limit(5).toList();
     }
 
     public void exportToCSV(File csvFile) {
+        if (data.isEmpty()) {
+            log.warn("No data has been generated, ignoring CSV export");
+            return;
+        }
         // Write all simulation data to CSV
         try (BufferedWriter writer = new BufferedWriter(new FileWriter(csvFile))) {
             // Write comprehensive header
-            writer.write("Simulation,Max Windspeed (m/s),Wind Direction (deg),Temperature (K),Pressure (mbar)," +
-                    "Initial Stability,Min Stability,Max Stability,Apogee Stability,Apogee (m)," +
-                    "Max Mach,Landing Latitude(° N), Landing Longtitude(° E)\n");
+            StringBuilder header = new StringBuilder("Simulation,Max Windspeed (m/s),Wind Direction (deg),Temperature (°C),Pressure (mbar),Apogee (ft),Max Mach");
+
+            String[] branchHeaders =
+                    {"Initial Stability", "Min Stability", "Max Stability", "Apogee Stability", "Landing Latitude (°N)",
+                            "Landing Longitude (°E)", "Position East of Launch (ft)", "Position North of Launch (ft)"};
+            int branches = data.get(0).getBranchName().size();
+            for (int i = 0; i < branches; i++) {
```

Comment:
from running it myself, I only see that there's one flight branch, so is there a need to do this? furthermore, if there were more than one flight branch, wouldn't there be duplicate headers? I think that would be a bit confusing